blog/_posts/2016-10-24-scalafix.md

```diff
@@ -0,0 +1,215 @@
+lazy val x = ... // before
+@volatile lazy val x = ... // after
+~~~
+
+With `@volatile`, Dotty uses a deadlock free scheme that is
+comparable-if-not-faster than the scheme used in scalac.
+
+How does scalafix work?
+---------------------------
+
+All the heavy-lifting behind scalafix happens in
+[scala.meta](http://scalameta.org/), a new metaprogramming toolkit,
+aspiring to succeed scala.reflect. Scala.meta makes it easy to do
+non-trivial code transformations while preserving syntactic details in
+the original source file. This key attribute of scala.meta makes it
+suitable for both developer tools like scalafix as well as compile-time
+metaprograms like macros.
+
+The grand vision of scala.meta is to provide a single interface to
+accomplish most common metaprogramming tasks in
+Scala.
+![scala.meta sketch]({{ site.baseurl }}/resources/img/scalameta-sketch.jpg)
+ The idea is to untie metaprograms from compiler implementations, providing a
+platform-independent API that works both in the Scala compiler, Dotty
+and tools like IntelliJ IDEA. Metaprogram authors benefit from a
+user-friendly interface built on immutable ADTs and type classes.
+Compiler authors benefit from the fact that they don't need to expose
+compiler internals in order to support popular features like macros.
+
+In scalafix, we use the scala.meta API to parse Scala source files,
+traverse abstract syntax trees, inspect tokens (including comments!) and
+then perform the minimal required transformation to the source file.
+Thanks to scala.meta dialects, which abstract away syntactic differences
+between variants of Scala, we can use the same programs to manipulate
+regular Scala source files, SBT files as well as code that uses new
+Dotty syntax such as union types and trait parameters. All this powerful
+functionality is implemented behind a minimalistic interface.
+The NonVolatileLazyVal
+[implementation](https://github.com/scalacenter/scalafix/blob/master/core/src/main/scala/scalafix/rewrite/VolatileLazyVal.scala)
+is 15 lines of code. It could comfortably fit in three tweets.
+
+We would not be able to achieve this level of expressiveness with
+scala.reflect, the current standard metaprogramming framework for Scala.
+Why? To name a few reasons, scala.reflect does not preserve syntactic
+details such as comments and scala.reflect has no notions of dialects,
+so it is bound to only support Scala 2.x.
+
+How does scalafix evolve?
+-----------------------------
+
+As we have seen above, the functionality of scalafix heavily relies on
+the infrastructure provided by scala.meta. However, in order to write
+more sophisticated scalafix rewrite rules, there are two main features
+missing in scala.meta, namely a *scalac converter* and *semantic API*.
+We are closely collaborating with Eugene Burmako, the lead developer of
+scala.meta, in the development efforts of these two features.
+
+### Scalac converter
+
+The scalac converter is a key subsystem of scala.meta. For every Scala
```

Comment:
Maybe show specifically where this is in the meta repo? Just in case people want to learn about the parts of meta and potentially brave a contribution.